kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -444,19 +525,59 @@ class KinematicsBase
                                    std::string* error_text_out = NULL) const;
 
   /**
-   * @brief  Set the search discretization
+   * @brief  Set the search discretization value for all the redundant joints
    */
   void setSearchDiscretization(double sd)
   {
-    search_discretization_ = sd;
+    redundant_joint_discretization_.clear();
+    for(std::vector<unsigned int>::iterator i = redundant_joint_indices_.begin();
+       i != redundant_joint_indices_.end();i++)
+    {
+      redundant_joint_discretization_[*i] = sd;
+    }
+  }
+
+  /**
+   * @brief Sets individual discretization values for each redundant joint.
+   *
+   * Calling this method replaces previous discretization settings.
+   *
+   * @param discretization a map of joint indices and discretization value pairs.
+   */
+  void setSearchDiscretization(const std::map<int,double>& discretization)
```

Comment:
Is it always up to the KinematicsBase implementation to set the redundant joints? If that's the case then we can check the keys in the input discretization map instead resetting the joints from it.